l2-contracts/contracts/bridge/AssetRouterBase.sol

```diff
@@ -0,0 +1,266 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+import {Ownable2StepUpgradeable} from "@openzeppelin/contracts-upgradeable/access/Ownable2StepUpgradeable.sol";
+import {PausableUpgradeable} from "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import {IAssetRouterBase} from "./interfaces/IAssetRouterBase.sol";
+import {IAssetHandler} from "./interfaces/IAssetHandler.sol";
+import {INativeTokenVault} from "./interfaces/INativeTokenVault.sol";
+import {IBridgehub, L2TransactionRequestTwoBridgesInner, L2TransactionRequestDirect} from "./interfaces/IBridgehub.sol";
+
+import {L2_NATIVE_TOKEN_VAULT, L2_ASSET_ROUTER, TWO_BRIDGES_MAGIC_VALUE} from "../L2ContractHelper.sol";
+
+/// @author Matter Labs
+/// @custom:security-contact security@matterlabs.dev
+/// @dev Bridges assets between L1 and ZK chain, supporting both ETH and ERC20 tokens.
+/// @dev Designed for use with a proxy for upgradability.
+abstract contract AssetRouterBase is IAssetRouterBase, Ownable2StepUpgradeable, PausableUpgradeable {
```

Comment:
Currently there seem to be two AssetRouterBase contracts. I think we should move the l2 bridge contracts to the l1-contracts folder. We already zk-compile there, so that part should be ok. Probably we could also run testing there, with another command?

Reply:
Ok sounds good. I have a few small things left to fix, and want to double-check circular interfaces, then will perform these steps